### ISSUE TITLE
training details about ByteScheduler

### ISSUE THREAD

**Issue by Rivendile** [bytedance/byteps#300]
I would like to reproduce the results shown in ByteScheduler. However, I didn't find the dataset used for these models and I can't decide some parameters, such as the image size and token numbers of a sample. Could you please tell me what setting should I use to reproduce the results?

**Comment by ymjiang** [comment]
@pengyanghua : Could you please introduce more about the experiment details?

**Comment by Rivendile** [comment]
Furthermore, I would appreciate it if you could provide step-by-step instruction to run the MXNet-image-classification example.

**Comment by pengyanghua** [comment]
@Rivendile Hi，The dataset for image classification and transformer is ImageNet 2012 and WMT17 respectively. For MXNet image classification in ByteScheduler, use the script here https://github.com/bytedance/byteps/blob/bytescheduler/bytescheduler/examples/mxnet-image-classification/train_imagenet.py. All used settings are default except that batch size is 32 and num_classes is 10000. Feel free to email me if you have any problems when reproducing the results.

**Comment by pengyanghua** [comment]
@Rivendile Hi, The dataset for image classification and transformer is ImageNet 2012 and WMT17 respectively. For MXNet image classification in ByteScheduler, use the script here https://github.com/bytedance/byteps/blob/bytescheduler/bytescheduler/examples/mxnet-image-classification/train_imagenet.py. All used settings are default except that batch size is 32 and num_classes is 10000. Feel free to email me if you have any problems when reproducing the results.

**Comment by Rivendile** [comment]
Hi, could you please tell me which code for Transformer did you use?

**Comment by pengyanghua** [comment]
@Rivendile Hi, we use the [tensor2tensor](https://github.com/tensorflow/tensor2tensor/tree/v1.13.1) libaray with hyper-parameters "--model=transformer --problem=translate_ende_wmt32k --hparams_set=transformer_base --worker_gpu=8 --save_checkpoints_secs=600000 --train_steps=1000 --std_server_protocol=grpc --log_step_count_steps=20 --hparams='batch_size=512"

**Comment by wydwww** [comment]
Hi @pengyanghua, can you please suggest how to train Transformer with MXNet (using Module API)? Thanks.

**Comment by pengyanghua** [comment]
@wydwww Hi, we used [sockeye](https://github.com/awslabs/sockeye/releases/tag/1.18.92) with arguments "--learning-rate-scheduler-type fixed-rate-inv-sqrt-t --kvstore dist_sync --batch-size 4096 -s wmt17/wmt17-train.de -t wmt17/wmt17-train.en -vs wmt17/wmt17-val.de -vt wmt17/wmt17-val.en -o wmt_model --device-ids -8 --measure-speed-every 20 --max-updates 1000 --encoder transformer --decoder transformer".

**Comment by Rivendile** [comment]
@pengyanghua Thanks a lot!

**Comment by pengyanghua** [comment]
@Rivendile 1. You may set a larger credit value if the number of workers increases. 2. They are two different libraries. You may check priority scheduling in byteps [here](https://github.com/bytedance/byteps/blob/bytescheduler/byteps/common/scheduled_queue.cc#L26).

**Comment by Rivendile** [comment]
Thanks for your reply. And I'll try as you suggested.
